src/routes/sessions/[id]/+page.svelte

```diff
@@ -109,6 +105,29 @@
 		await handleCompletion(payload);
 	}
 
+	async function handleSubmitEditMessage() {
+		if (messageIndexToEdit === null) return;
+
+		session.messages[messageIndexToEdit].content = prompt;
+
+		// Remove all messages after the edited message
+		session.messages = session.messages.slice(0, messageIndexToEdit + 1);
+
+		let payload = { model: session.model, messages: session.messages, stream: true };
+		await handleCompletion(payload);
+
+		messageIndexToEdit = null;
+		prompt = '';
+	}
+
+	$: handleSubmit = () => {
+		if (!prompt) return;
+		isPromptFullscreen = false;
+
+		if (messageIndexToEdit !== null) handleSubmitEditMessage();
+		else handleSubmitNewMessage();
+	};
```

Comment:
This feels like a code smell, I don't think we need a reactive function to handle this.
We can simplify it by moving the logic to `<ButtonSubmit>`. See note below.